builtin/logical/database/backend.go

```diff
@@ -85,13 +85,11 @@ func Backend(conf *logical.BackendConfig) *databaseBackend {
 	b.logger = conf.Logger
 	b.connections = make(map[string]*dbPluginInstance)
 
-	// Always populate the queue to guard against nil pointers.
-	b.credRotationQueue = queue.NewTimeQueue()
-
-	// But only execute cred rotation if we're the leader.
-	// TODO what if leadership changes during the lifecycle of the application?
 	replicationState := conf.System.ReplicationState()
-	if replicationState.IsLeader() {
+	if b.System().LocalMount() || !replicationState.IsFollower() {
```

Comment:
This needs to be (local mount || !performancesecondary) && !drsecondary && !perfstandby

Reply:
This needs to be (local mount || !performancesecondary) && !drsecondary && !perfstandby
👍 1

Reply:
Thank you! Super helpful.